src/Bundle/Provider/ServiceGridProvider.php

```diff
@@ -40,6 +48,21 @@ public function get(string $code): Grid
 
         $gridConfiguration = $grid->toArray();
 
+        if (isset($gridConfiguration['extends'])) {
+            $gridConfiguration = $this->extend($gridConfiguration, $gridConfiguration['extends']);
+        }
+
         return $this->converter->convert($code, $gridConfiguration);
     }
+
+    private function extend(array $gridConfiguration, string $parentGridCode): array
+    {
+        $parentGrid = $this->gridRegistry->getGrid($parentGridCode);
+
+        Assert::notNull($parentGrid, sprintf('Parent grid with code "%s" does not exists.', $parentGridCode));
+
+        $parentGridConfiguration = $parentGrid->toArray();
+
+        return $this->gridConfigurationExtender->extends($gridConfiguration, $parentGridConfiguration);
+    }
```

Comment:
Can't we extract the whole part to separate service? Therefore, we would throw an exception only in one place - and for whatever reason it would be easier for others to replace/change this logic

Reply:
You mean to only pass the parent grid code to the extender and use the grid registry to get the grid by code and throw exception if it doesn't exists?

Reply:
Something like this. The content of this method seems to be almost duplicated and I assume, we would like to use the same logic if we will need to use this function again

Reply:
It's similar, but not the same.
Here, the grid registry is used, while the ArrayGridProvider uses an array copy of grid configuration, that's where it gets it's name from.
@loic425 are those two compatible? Can a grid registered as service extend a grid from config? What about the other way around?

Reply:
In theory, with "extends" method you can extend a grid from config with a grid registered as service but not the other way around.

The solution on the service provider is to look on the grid registry to find the grid configuration. The grid registry has its providers sorted by priority (by default, ServiceGridProvider is on first position).

It's impossible for now, cause GridRegistry is on the bundle part and the ArrayGridProvider is on Component part.

Reply:
At first, I added all these services into Component :D and made the DI into the bundle part.

Reply:
Let's address this separately.

Reply:
@lchrusciel Ok for you too?

Reply:
Ok, I'm fine with that 👍